db/go_level_db.go

```diff
@@ -156,17 +169,31 @@ func (mBatch *goLevelDBBatch) Write() {
 //----------------------------------------
 // Iterator
 
-func (db *GoLevelDB) Iterator() Iterator {
+func (db *GoLevelDB) Iterator(start, end []byte) Iterator {
 	itr := &goLevelDBIterator{
 		source: db.db.NewIterator(nil, nil),
+		start:  start,
+		end:    end,
 	}
-	itr.Seek(nil)
+	itr.source.Seek(start)
```

Comment:
what about the same check above?

```
if len(start) > 0 {
  itr.source.Seek(start)
} else {
  itr.source.SeekToFirst()
}
```

Reply:
C-LevelDB requires that the key itself is not empty, though it's not documented, so you can't pass an empty key: https://github.com/jmhodges/levigo/blob/master/iterator.go#L127

GoLevelDB seems to be fine with empty keys. But you're right, would probably be better to check.